src/parse.h

```diff
@@ -19,6 +19,12 @@
 
 namespace neml {
 
+/// Parse from a string to a shared_ptr
+std::shared_ptr<NEMLModel> parse_string(std::string input);
```

Comment:
can we rename this as
parse_string -> parse_string_shared_ptr
parse_string_unique -> parse_string_unique_ptr

or if the pointer is implied
at least
parse_string -> parse_string_shared

Reply:
Ehh, I decline for now to maintain consistency with the current XML *file* parsing naming convention.  If we change the file parser that propagates to a lot of other locations.  If you want to make a separate pull request to do that be my guest.